spirv_msl.cpp

```diff
@@ -11608,11 +11608,15 @@ string CompilerMSL::to_struct_member(const SPIRType &type, uint32_t member_type_
 		}
 	}
 
-	// Very specifically, image load-store in argument buffers are disallowed on MSL on iOS.
-	if (msl_options.is_ios() && physical_type.basetype == SPIRType::Image && physical_type.image.sampled == 2)
+	// iOS Tier 1 argument buffers do not support writable images.
+	if (physical_type.basetype == SPIRType::Image &&
+		physical_type.image.sampled == 2 &&
+		msl_options.is_ios() &&
+		msl_options.argument_buffers &&
```

Comment:
The argument buffer check seems a bit redundant here. The tier check should be enough. Is it possible to place images in structs without those?

Reply:
Hmmm. I added it because there was nothing in the `to_struct_member()` function that indicates argument buffers are in use. But I guess you implying that the only reason an image would be part of a struct is when argument buffers are being used?

I guess that makes sense, and I may have overlooked that. 

Fixed.